From b00c5116a41f106737f011b4c803b7451941a732 Mon Sep 17 00:00:00 2001
Message-Id: <b00c5116a41f106737f011b4c803b7451941a732.1387382496.git.minovotn@redhat.com>
In-Reply-To: <c5386144fbf09f628148101bc674e2421cdd16e3.1387382496.git.minovotn@redhat.com>
References: <c5386144fbf09f628148101bc674e2421cdd16e3.1387382496.git.minovotn@redhat.com>
From: Nigel Croxon <ncroxon@redhat.com>
Date: Thu, 14 Nov 2013 22:53:09 +0100
Subject: [PATCH 33/46] rdma: silly ipv6 bugfix

RH-Author: Nigel Croxon <ncroxon@redhat.com>
Message-id: <1384469598-13137-34-git-send-email-ncroxon@redhat.com>
Patchwork-id: 55715
O-Subject: [RHEL7.0 PATCH 33/42] rdma: silly ipv6 bugfix
Bugzilla: 1011720
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>
RH-Acked-by: Amit Shah <amit.shah@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

Bugzilla: 1011720
https://bugzilla.redhat.com/show_bug.cgi?id=1011720

>From commit ID:
commit c89aa2f1851b08c3efa8a1070c0a6b9a36e1227f
Author: Michael R. Hines <mrhines@us.ibm.com>
Date:   Sun Aug 18 22:27:08 2013 -0400

    rdma: silly ipv6 bugfix

    My bad - but it's very important for us to warn the user that
    IPv6 is broken on RoCE in linux right now, until linux releases
    a fixed version.

    Signed-off-by: Michael R. Hines <mrhines@us.ibm.com>
    Signed-off-by: Michael Tokarev <mjt@tls.msk.ru>
---
 migration-rdma.c |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 migration-rdma.c | 8 +++++---
 1 file changed, 5 insertions(+), 3 deletions(-)

diff --git a/migration-rdma.c b/migration-rdma.c
index ada488e..943496f 100644
--- a/migration-rdma.c
+++ b/migration-rdma.c
@@ -920,9 +920,11 @@ static int qemu_rdma_resolve_host(RDMAContext *rdma, Error **errp)
         ret = rdma_resolve_addr(rdma->cm_id, NULL, e->ai_dst_addr,
                 RDMA_RESOLVE_TIMEOUT_MS);
         if (!ret) {
-            ret = qemu_rdma_broken_ipv6_kernel(errp, rdma->cm_id->verbs);
-            if (ret) {
-                continue;
+            if (e->ai_family == AF_INET6) {
+                ret = qemu_rdma_broken_ipv6_kernel(errp, rdma->cm_id->verbs);
+                if (ret) {
+                    continue;
+                }
             }
             goto route;
         }
-- 
1.7.11.7

